auth/deployment.yaml

```diff
@@ -0,0 +1,110 @@
+apiVersion: apps/v1beta2
```

Comment:
I think we should have component deployment tests. Could be part of a larger integrated test.

Reply:
What's a component deployment test?

Reply:
Test that the component/service can be deployed successfully to Kubernetes

Reply:
I'm still not sure I understand.  All the components are deployed during the tests.

Reply:
I thought auth wasn't being tested (the integrated test would be what you described), I think because I was expecting it to be tested directly. So batch is testing auth endpoints, and we specify the dependency to deploy auth in the testing namespace. 

I do see a deploy_auth dependency in deploy_batch, which makes deploy_auth a transitive dependency of

```sh
 - kind: deploy
   name: test_batch
   namespace:
     valueFrom: batch_pods_ns.name
   config: batch/test-batch-pod.yaml
   wait:
    - kind: Pod
      name: test-batch
      for: completed
      timeout: 900
   dependsOn:
    - default_ns
    - batch_pods_ns
    - deploy_batch
    - deploy_batch_pods
    - test_batch_image
```

That wasn't immediately obvious to me, sorry.

Reply:
> I thought auth wasn't being tested

No problem, happy to clarify.  Auth is being partially tested.  Auth is now a lynchpin and nothing can function without it.  Auth is being deployed in tests and used by all the clients and services for authorization.  The login/logout flow is not being tested.  The way we get accounts without login is by creating sessions directly in the create_accounts step and saving them to <user>-tokens secrets (like we did with JWTs before).